### PR TITLE
Groups form fields

### LINKED ISSUE #141: Group fields into sections based on metadata type
The metadata type is part of the metadata spreadsheet Emily gave us.

- [ ] Group descriptive metadata in a section at the top of the form.
- [ ] Group rights metadata in a section in the middle.
- [ ] Group admin metadata in a section at the bottom.

### PR DESCRIPTION
**Pull request by devanshu-m**
This commit groups form metadata fields. Also adds `xray-rails` gem.

Closes #141 

image

image

### CODE REVIEW

File: spec/features/create_curate_generic_work_spec.rb [hunk 1]
```diff
@@ -36,9 +36,12 @@
       expect(page).to have_css('#metadata input#curate_generic_work_title')
       expect(page).to have_css('#metadata select#curate_generic_work_rights_statement')
 
-      click_on 'Additional fields'
-
+      click_link('Additional descriptive fields')
       expect(page).to have_content('Add another Content genre')
+
+      sleep(2)
```
Comment (rotated8) [inline spec/features/create_curate_generic_work_spec.rb]: Is this necessary? Could we use a `wait` instead?

File: app/views/hyrax/base/_form_metadata.html.erb [hunk 2]
```diff
@@ -0,0 +1,78 @@
+<div class="descriptive-metadata">
+  <h4><%= t("hyrax.works.form.descriptive_metadata_heading") %></h4>
+
+  <div class="base-terms">
+    <% f.object.primary_descriptive_metadata_fields.each do |term| %>
+      <%= render_edit_field_partial(term, f: f) %>
+    <% end %>
+  </div>
+  <% if f.object.display_additional_fields? %>
```
Comment (rotated8) [inline app/views/hyrax/base/_form_metadata.html.erb]: Is there a way we can avoid repeating this section?

File: spec/features/create_curate_generic_work_spec.rb (hunk 1 above)
Comment (devanshu-m) [reply spec/features/create_curate_generic_work_spec.rb]: I will change this to `wait` 👍

File: app/views/hyrax/base/_form_metadata.html.erb (hunk 2 above)
Comment (devanshu-m) [reply app/views/hyrax/base/_form_metadata.html.erb]: We could call another partial from this one?

File: spec/features/create_curate_generic_work_spec.rb [hunk 3]
```diff
@@ -36,9 +36,10 @@
       expect(page).to have_css('#metadata input#curate_generic_work_title')
       expect(page).to have_css('#metadata select#curate_generic_work_rights_statement')
 
-      click_on 'Additional fields'
-
+      click_link('Additional descriptive fields')
       expect(page).to have_content('Add another Content genre')
+
+      click_link('Additional admin fields', wait: 2)
```
Comment (rotated8) [inline spec/features/create_curate_generic_work_spec.rb]: Can you add a comment about why the wait is necessary?

**Pull request merged**